### PR TITLE
[TEVA-1291] Weekly job alerts

### PR DESCRIPTION
**Pull request by cpjmcquillan**
The PR allows weekly job alerts to be sent. The code involved in the `Job alerts` journey is quite convoluted and could do with some TLC and refactoring to simplify things, but I haven't got the capacity to do it. Feel free to hijack this branch, or merge and refactor things where you see fit as part of any of the other job alert tickets. 

**TODO**
- [ ] Add weekly alert email job spec
- [ ] Update daily alert email job spec to test the daily nature
- [ ] Make it all nicer

## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1291

## Changes in this PR
- Add weekly alerts job
- Add weekly alerts queue
- Remove reference field from template
- Remove old feature flag

### CODE REVIEW

File: app/components/jobseekers/search_results/job_alerts_link_component.rb [hunk 1]
```diff
@@ -4,6 +4,6 @@ def initialize(vacancies_search:)
   end
 
   def render?
-    @vacancies_search.any? && EmailAlertsFeature.enabled? && !ReadOnlyFeature.enabled?
```
Comment (david-mears-dfe) [inline app/components/jobseekers/search_results/job_alerts_link_component.rb:7]: Can you also remove the flags from `terraform/workspace-variables`?

File: app/jobs/alert_email/base.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def perform
+    return unless Rails.env.production?
+
+    subscriptions.each do |s|
+      next if s.alert_run_today?
+
+      vacancies = vacancies_for_subscription(s)
+      next if vacancies.blank?
+
+      AlertMailer.alert(s.id, vacancies.pluck(:id)).deliver_later(queue: :email_alerts)
```
Comment (david-mears-dfe) [inline app/jobs/alert_email/base.rb:11]: `s` is hard to read - cognitive overload. I prefer `subscription`.
It's like when as a new joiner to the team I join a TVS meeting with the DM, PM, UX, and PA about MAT/DSI issues.

Comment (david-mears-dfe) [reply app/jobs/alert_email/base.rb:11]: It wasn't your original code I realise

File: config/sidekiq.yml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 :max_retries: 5
 
 :queues:
-  - [email_daily_alerts, 4]
+  - [email_alerts, 4]
```
Comment (david-mears-dfe) [inline config/sidekiq.yml:4]: schedule.yml says `daily_alerts`, this says `email_alerts`. I think they should be the same.

**Review by david-mears-dfe — APPROVED** [review]

**Review by david-mears-dfe — CHANGES_REQUESTED** [review]
Switching to 'request changes' because I can't see how to revoke my approval (as this PR is waiting for tests and things before it's ready)

**Review by david-mears-dfe — CHANGES_REQUESTED** [review]
This PR does not appear to change the text on the confirmation page or the confirmation email. I think those things have been missed, in between this PR and https://github.com/DFE-Digital/teacher-vacancy-service/pull/2055

File: app/mailers/alert_mailer.rb [hunk 4]
```diff
@@ -1,24 +1,19 @@
...
-      end
+    template = @subscription.daily? ? NOTIFY_SUBSCRIPTION_DAILY_TEMPLATE : NOTIFY_SUBSCRIPTION_WEEKLY_TEMPLATE
 
     view_mail(
-      NOTIFY_SUBSCRIPTION_DAILY_TEMPLATE,
+      template,
       to: @subscription.email,
-      subject: I18n.t(subject_key, reference: @subscription.reference),
+      subject: I18n.t("job_alerts.alert.email.#{@subscription.frequency}.subject",
+                      count: @vacancies.count, reference: @subscription.reference),
```
Comment (alexbowen) [inline app/mailers/alert_mailer.rb:16]: the reference can be removed now cant it?

Comment (alexbowen) [reply app/mailers/alert_mailer.rb:16]: actually no its in my other email PR, doent make sense to remove it here

**Review by alexbowen — APPROVED** [review]
i think looks ok to me, couple of thinking out loud comments, so approving

File: app/components/jobseekers/search_results/job_alerts_link_component.rb [hunk 5]
```diff
@@ -5,6 +5,6 @@ def initialize(vacancies_search:, count:)
   end
 
   def render?
-    @vacancies_search.any? && EmailAlertsFeature.enabled? && !ReadOnlyFeature.enabled? && @count.positive?
```
[the remark this answers is not in the document]
Comment (alexbowen) [reply app/components/jobseekers/search_results/job_alerts_link_component.rb:8]: i see you have removed feature flag below

File: spec/jobs/send_weekly_alert_email_job_spec.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  end
+
+  context 'with vacancies' do
+    before do
+      allow_any_instance_of(described_class).to receive(:vacancies_for_subscription) { vacancies }
+      allow(Rails).to receive(:env).and_return(ActiveSupport::StringInquirer.new('production'))
+    end
+
+    it 'sends an email' do
+      expect(Subscription).to receive(:weekly) { Subscription.where(frequency: :weekly) }
```
Comment (david-mears-dfe) [inline spec/jobs/send_weekly_alert_email_job_spec.rb:36]: Why do we need to specify this? Does the FactoryBot subscription not respond to `weekly` like the real subscription?

File: spec/controllers/subscriptions_controller_spec.rb [hunk 7]
```diff
@@ -8,77 +8,52 @@
   describe '#new' do
     subject { get :new, params: { search_criteria: { subject: 'english' } } }
```
Comment (david-mears-dfe) [inline spec/controllers/subscriptions_controller_spec.rb:9]: We used to have a subjects filter, but now we don't, so this should probably be a keyword, not a subject.

File: spec/jobs/send_weekly_alert_email_job_spec.rb (hunk 6 above)
Comment (cesidio) [reply spec/jobs/send_weekly_alert_email_job_spec.rb:36]: Ideally we should just do `expect(Subscription).to receive(:weekly)` but for some reasons, when the code is executed `Subscription.weekly` returns nil. So I had to explicitly set what it should return in the spec

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**